types/react/test/index.ts

```diff
@@ -117,16 +117,19 @@ declare const container: Element;
 
 class ModernComponent extends React.Component<Props, State, Snapshot>
     implements MyComponent, React.ChildContextProvider<ChildContext> {
-    static propTypes: React.ValidationMap<Props> = {
-        foo: PropTypes.number
+    static propTypes: PropTypes.ValidationMap<Props> = {
```

Comment:
It's worth noting that this should be tested both with and without the type annotation, ideally.